src/AutoRest.CSharp/Common/Output/Models/CmcRestClientBuilder.cs

```diff
@@ -302,7 +303,8 @@ protected virtual Parameter[] BuildMethodParameters(IReadOnlyDictionary<InputPar
                         // This method has a flattened body
                         if (bodyRequestParameter.Kind == InputOperationParameterKind.Flattened)
                         {
-                            var objectType = (SchemaObjectType)_library.ResolveModel((InputModelType)bodyRequestParameter.Type).Implementation;
+                            (InputType bodyType, bool isNullable) = bodyRequestParameter.Type is InputNullableType nullableType ? (nullableType.Type, true) : (bodyRequestParameter.Type, false);
+                            var objectType = (SchemaObjectType)_library.ResolveModel((InputModelType)bodyType, isNullable).Implementation;
```

Comment:
this is really unnecessary.
Our TypeFactory.CreateType could handle this perfectly.
Also since we only need its TypeProvider, and type provider does not have a concept of nullable, we do not need the extra parameter.

The best solution is that i think we should do this:
```csharp
var csharpType = _typeFactory.CreateType(bodyRequestParameter.Type); // this automatically handles nullable for you
Debug.Assert(csharpType is {IsFrameworkType: false, Implementation: SchemaObjectType objectType});
```